src/Common/Thermodynamics/relations.jl

```diff
@@ -904,7 +904,7 @@ function supersaturation(
 ) where {FT <: Real}
 
     q_sat::FT = q_vap_saturation_generic(param_set, T, ρ, Liquid())
-    q_vap::FT = q.tot - q.liq - q.ice
+    q_vap::FT = max(FT(0), q.tot - q.liq - q.ice)
```

Comment:
```suggestion
    q_vap = vapor_specific_humidity(q)
```
Then we can use `max(0, q.tot - q.liq - q.ice)` in `vapor_specific_humidity`.

Reply:
I was doing exactly that. I'll have a separate PR with just those `relations.jl` fixes.